### PR TITLE
Add docker files

### PR DESCRIPTION
**Pull request by anaion**
PR for #26735

@alexef please review

### CODE REVIEW

File: src/nbsap/settings.py.docker [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+ASSETS_ROOT = os.path.join(BASE_DIR, 'static')
+
+# Main config
+EU_STRATEGY = False
+EU_STRATEGY_ADD = False
+NAT_STRATEGY = True
+SITE_HEADER = 'NBSAP'
+INFO_HEADER = False
+LAYOUT_FOOTER_LOGO_VISIBLE = False
+LAYOUT_HEADER_LOGO_VISIBLE = False
```
Comment (alexef) [inline src/nbsap/settings.py.docker]: Please take EU_STRATEGY, EU_STRATEGY_ADD, NAT_STRATEGY, SITE_HEADER, INFO_HEADER from the environment.

File: src/nbsap/settings.py.docker [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+TEST_RUNNER = 'django_nose.NoseTestSuiteRunner'
+NOSE_ARGS = [
+    '--with-coverage',
+    '--cover-package=nbsap',
+]
+
+# Override
+if not DEBUG:
+    SENTRY_DSN = 'http://e58d6f13a36d419d81c1d1e44e5e2519:d67d512c5f4343c0aebcdca30bc64871@sentry.mojito.edw.ro/16'
```
Comment (alexef) [inline src/nbsap/settings.py.docker]: Take it from the environment, default to ''.

File: src/nbsap/settings.py.docker [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+try:
+    from local_settings import *
+except ImportError:
+    pass
+
+if 'test' in sys.argv:
+    try:
+        from test_settings import *
+    except ImportError:
+        pass
```
Comment (alexef) [inline src/nbsap/settings.py.docker]: Last 10 lines not needed in the image.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#start from an ubuntu image
+FROM ubuntu
```
Comment (alexef) [inline Dockerfile]: Use a tag on ubuntu, like 14.04, don't use latest, it may break in the future.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+#add the code to the image
+RUN mkdir /nbsap
+WORKDIR /nbsap
+ADD . /nbsap
+
+#install the requirements
+RUN pip install -r requirements-dev.txt
+RUN pip install -U distribute
+RUN pip install -e .
```
Comment (alexef) [inline Dockerfile]: Add a CMD statement for the default entrypoint.

python instance/manage.py runserver 0.0.0.0:8000

Comment (alexef) [inline Dockerfile]: Expose 8000 with the ENV directive, so it can be easily overwritten.

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    command:
+        python instance/manage.py runserver 0.0.0.0:8000
+    volumes:
+        - .:/nbsap
+    ports:
+        - "8000:8000"
+    links:
+        - db
+    environment:
+        - DEBUG=True
```
Comment (alexef) [inline docker-compose.yml]: Please move this file to a different repository, see: https://taskman.eionet.europa.eu/projects/infrastructure/wiki/Docker_image_creation

I have created: https://github.com/eea/eea.docker.nbsap

This docker-compose.yml should be PR-ed there as a docker-compose.dev.yml. 

Another one with production setup (replicating demo, ec and country instances currently on calamar should be added).

File: Dockerfile (hunk 5 above)
Comment (anaion) [inline Dockerfile]: I don't agree with adding the CMD in the Dockerfile. The image should only contain static data, as it's also used for other management operations such as syncdb, load_fixtures etc.

File: src/nbsap/settings.py.docker [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+
+MANAGERS = ADMINS
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.mysql', # Add 'postgresql_psycopg2', 'mysql', 'sqlite3' or 'oracle'.
+        'NAME': 'nbsap',                      # Or path to database file if using sqlite3.
+        # The following settings are not used with sqlite3:
+        'USER': 'nbsap',
+        'PASSWORD': 'nbsap',
```
Comment (alexef) [inline src/nbsap/settings.py.docker]: For the production setup, we will want to take the database configuration from the environment.

File: src/nbsap/settings.py.docker [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+# http://www.i18nguy.com/unicode/language-identifiers.html
+LANGUAGE_CODE = 'en'
+
+ugettext = lambda s: s
+
+LANGUAGES = (
+    ('en', ugettext('English')),
+    ('fr', ugettext('French')),
+    ('nl', ugettext('Dutch')),
+)
```
Comment (alexef) [inline src/nbsap/settings.py.docker]: These will also be different from one country instance to another.

Maybe mount the `local_settings.py` as a volume to override them is a good solution? What do you think?

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,15 @@
+#start from an ubuntu image
+FROM ubuntu
+RUN apt-get -y update
+RUN apt-get -y install python2.7 python2.7-dev python-setuptools python-pip
+RUN apt-get -y install libldap2-dev libsasl2-dev libmysqlclient-dev libxml2-dev libxslt1-dev
+
+#add the code to the image
+RUN mkdir /nbsap
+WORKDIR /nbsap
+ADD . /nbsap
```
Comment (alexef) [inline Dockerfile]: ADD settings.py.docker /nbsap/instance/settings.py ?

File: src/nbsap/settings.py.docker (hunk 8 above)
Comment (anaion) [inline src/nbsap/settings.py.docker]: Isn't it automatically overridden when mounting .?
+ volumes:
+        - .:/nbsap

File: docker-compose.yml (hunk 6 above)
Comment (anaion) [inline docker-compose.yml]: Can you please give me more details about the last line?

File: src/nbsap/settings.py.docker (hunk 8 above)
Comment (alexef) [inline src/nbsap/settings.py.docker]: This won't be the case in production.

File: Dockerfile (hunk 5 above)
Comment (alexef) [inline Dockerfile]: Adding a CMD directive has nothing to do with using the image for other management operations. The CMD is used as a default command, when no specific command is passed.

Please see: https://docs.docker.com/reference/builder/#cmd

File: docker-compose.yml (hunk 6 above)
Comment (alexef) [inline docker-compose.yml]: In https://github.com/eea/eea.docker.nbsap you should put two docker-compose files:

- this file, but renamed `docker-compose.dev.yml`, to be used as a development setup
- a new one, named `docker-compose.yml` with containers setup for the instances on calamar (the country instances + demo + ec)

File: docker-compose.yml [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+        - DB_NAME='nbsap'
+        - DB_USER='nbsap'
+        - DB_PASSWORD='nbsap'
+        - DB_HOST='db'
+        - SENTRY_DSN='http://e58d6f13a36d419d81c1d1e44e5e2519:d67d512c5f4343c0aebcdca30bc64871@sentry.mojito.edw.ro/16'
+        - EU_STRATEGY=True
+        - EU_STRATEGY_ADD=True
+        - NAT_STRATEGY=False
+        - SITE_HEADER='NBSAP'
+        - INFO_HEADER=False
```
Comment (alexef) [inline docker-compose.yml]: It's better practice to use an environment file for such amount of environment variables, please see: https://docs.docker.com/compose/yml/#env-file

File: Dockerfile (hunk 5 above)
Comment (anaion) [inline Dockerfile]: Got it, thanks for the info.

**Comment by alexef** [comment]
Ok to merge.

**Pull request merged**